LeaderboardBackend/Controllers/UsersController.cs

```diff
@@ -1,86 +1,75 @@
-using Microsoft.AspNetCore.Mvc;
-using LeaderboardBackend.Models;
 using LeaderboardBackend.Controllers.Requests;
-using BCryptNet = BCrypt.Net.BCrypt;
-using Microsoft.AspNetCore.Authorization;
+using LeaderboardBackend.Models;
 using LeaderboardBackend.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using BCryptNet = BCrypt.Net.BCrypt;
+
+namespace LeaderboardBackend.Controllers;
 
-namespace LeaderboardBackend.Controllers
+[Route("api/[controller]")]
+[ApiController]
+public class UsersController : ControllerBase
 {
-	[Route("api/[controller]")]
-	[ApiController]
-	public class UsersController : ControllerBase
+	private readonly IUserService _userService;
+	private readonly IAuthService _authService;
+
+	public UsersController(IUserService userService, IAuthService authService)
+	{
+		_userService = userService;
+		_authService = authService;
+	}
+
+	[HttpGet("{id}")]
+	public async Task<ActionResult<User>> GetUser(Guid id)
 	{
-		private readonly IUserService _userService;
-		private readonly IAuthService _authService;
+		return await _userService.GetUser(id) is User user
+			? Ok(user)
+			: NotFound();
+	}
 
-		public UsersController(
-			IUserService userService,
-			IAuthService authService
-		)
+	[AllowAnonymous]
+	[HttpPost("register")]
+	public async Task<ActionResult<User>> Register([FromBody] RegisterRequest body)
+	{
+		if (body.Password != body.PasswordConfirm)
 		{
-			_userService = userService;
-			_authService = authService;
+			return BadRequest();
 		}
 
-		[HttpGet("{id}")]
-		public async Task<ActionResult<User>> GetUser(Guid id)
+		var newUser = new User
 		{
-			User? user = await _userService.GetUser(id);
-			if (user == null)
-			{
-				return NotFound();
-			}
+			Username = body.Username,
+			Email = body.Email,
+			Password = BCryptNet.EnhancedHashPassword(body.Password)
+		};
 
-			return Ok(user);
-		}
+		await _userService.CreateUser(newUser);
+		return CreatedAtAction(nameof(GetUser), new { id = newUser.Id }, newUser);
+	}
 
-		[AllowAnonymous]
-		[HttpPost("register")]
-		public async Task<ActionResult<User>> Register([FromBody] RegisterRequest body)
+	[AllowAnonymous]
+	[HttpPost("login")]
+	public async Task<ActionResult<User>> Login([FromBody] LoginRequest body)
+	{
+		if (await _userService.GetUserByEmail(body.Email) is not User user)
```

Comment:
Similar to above, this seems a lot less readable than what was here before. I prefer the old style of null checking.